lib/simple_config.rb

```diff
@@ -75,7 +75,10 @@ def get(key)
     # @return The current value for +:key+.
     def unset(key)
       singleton_class.send(:remove_method, key)
-      @settings.delete(key)
+      setting = @settings.delete(key)
+      # If there was a setting return that, otherwise continue to try and remove a group which may exist
+      return setting unless setting.nil?
```

Comment:
I can do 

```set :foo, ''```

which is a valid thing to set (thinking beyond our use in the pipeline)